pymtl3/stdlib/test/stateful/test_stateful_queues_test.py

```diff
@@ -0,0 +1,302 @@
+#=========================================================================
+# test_stateful_qeueues_test
+#=========================================================================
+# Tests for test_stateful using single entry queue
+#
+# Author: Yixiao Zhang
+#   Date: June 10, 2019
+
+from __future__ import absolute_import, division, print_function
+
+from collections import deque
+
+import pytest
+
+from pymtl3 import *
+from pymtl3.dsl.test.sim_utils import simple_sim_pass
+from pymtl3.passes import GenDAGPass, OpenLoopCLPass
+from pymtl3.stdlib.ifcs import CalleeIfcRTL
+
+from .test_stateful import get_strategy_from_type, run_test_state_machine
+from .test_wrapper import *
+
+
+#-------------------------------------------------------------------------
+# SingleEntryPipeQueue
+#-------------------------------------------------------------------------
+class SingleEntryPipeQueue( Component ):
+
+  def construct( s, EntryType ):
+
+    # Interface
+
+    s.enq = CalleeIfcRTL( ArgTypes=[( 'msg', EntryType ) ] )
```

Comment:
I was thinking if we should create the BitStruct here, instead of doing that in the interface, but maybe this looks better? @yo96 what do you think?

Reply:
I think this is a way to enforce that args and rets are BitStruct. Otherwise, a user can pass in anything like Bits

Reply:
hmm maybe they could pass in a Bits object if they want? Does it change anything?

Reply:
the adapter is implemented based on the assumption that args is BitsStruc. And in stateful testing, it is assumed that rets is BitsStruct whe comparing results. https://github.com/cornell-brg/pymtl3/blob/c09badc088611bc13bf19aeb92fe669c5664b614/pymtl3/stdlib/ifcs/CalleeIfc.py#L85-L89

Reply:
Not sure if this is the best way to do it: I made CalleeIfcRTL, CallerIfcRTL and CalleeRTL2CL take ArgType and RetType directly and added two helper function to create the BitStruct: https://github.com/cornell-brg/pymtl3/blob/c5568e5735a171245d613e4a141f8fb8bceea5e0/pymtl3/stdlib/ifcs/generic_method_ifcs.py#L170-L199  In this way the user can just do https://github.com/cornell-brg/pymtl3/blob/c5568e5735a171245d613e4a141f8fb8bceea5e0/pymtl3/stdlib/test/stateful/test_stateful_queues_test.py#L33-L34 which will 1. look clean 2. Enforce that the types passed into CalleeIfcRTL are BitsStruct so that adapter and connect could work as expected